### ISSUE TITLE
Elastic Beanstalk Solution Stack Name stale in State

### ISSUE THREAD

**Issue by tecnobrat** [hashicorp/terraform#7300]
Running terraform 0.6.16.

In my statefile stored on atlas, it has: `"solution_stack_name": "64bit Amazon Linux 2016.03 v2.1.1 running Multi-container Docker 1.9.1 (Generic)"`

However when I query the same environment using AWS's API, it is still on `v2.1.0`

```
[~]$ aws --region us-east-1 elasticbeanstalk describe-environments --application-name tf-rise-runtime
{
    "Environments": [
        {
            "ApplicationName": "tf-rise-runtime",
            "EnvironmentName": "rise-runtime",
            "VersionLabel": "0f37b4c81b01d3e83c1162b86595a3e9bb1bdf72-jenkins-eb-deploy-stage-1152",
            "Status": "Ready",
            "EnvironmentLinks": [],
            "EnvironmentId": "e-3m3iciyvzs",
            "EndpointURL": "awseb-e-3-AWSEBLoa-S3Z7AKU3CQJL-583246180.us-east-1.elb.amazonaws.com",
            "SolutionStackName": "64bit Amazon Linux 2016.03 v2.1.0 running Multi-container Docker 1.9.1 (Generic)",
            "CNAME": "rise-runtime.yuh2qyr258.us-east-1.elasticbeanstalk.com",
            "Health": "Green",
            "AbortableOperationInProgress": false,
            "Tier": {
                "Version": " ",
                "Type": "Standard",
                "Name": "WebServer"
            },
            "HealthStatus": "Ok",
            "DateUpdated": "2016-06-22T15:43:54.439Z",
            "DateCreated": "2016-06-01T23:01:17.618Z"
        }
    ]
}
```

As you can see, the solution stack name is actually `"SolutionStackName": "64bit Amazon Linux 2016.03 v2.1.0 running Multi-container Docker 1.9.1 (Generic)"`

I do not know why terraform isn't trying to keep this in sync and update it to `2.1.1`

**Comment by dharrisio** [comment]
My suspicion here is that this was caused by the solution stack attribute not being read in the environment read function. If there was an error when applying some change to the beanstalk environment the state file may have the updated solution stack stored in its state, and is unaware that beanstalk rolled back the failed apply.

@tecnobrat do you remember running into anything like that? Just waiting for tests to finish, but once that is done I will submit a PR that hopefully fixes this.

**Comment by tecnobrat** [comment]
This sounds about right. May have been a timeout.

On Thursday, 30 June 2016, David Harris <notifications@github.com> wrote:

> My suspicion here is that this was caused by the solution stack attribute
> not being read in the environment read function. If there was an error when
> applying some change to the beanstalk environment the state file may have
> the updated solution stack stored in its state, and is unaware that
> beanstalk rolled back the failed apply.
>
> @tecnobrat <https://github.com/tecnobrat> do you remember running into
> anything like that? Just waiting for tests to finish, but once that is done
> I will submit a PR that hopefully fixes this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashicorp/terraform/issues/7300#issuecomment-229811169>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAI3_txZUpP7M4Z1P-xoVWw1sl7spvOOks5qREjMgaJpZM4I9BP6>
> .
>


-- 
- Brian

**Comment by tecnobrat** [comment]
@stack72 can we open this again due to the fact that the PR was reverted?

**Comment by dharrisio** [comment]
@tecnobrat From your original comment for the `solution_stack_name` value in your state file: was that from the config template resource or the environment resource?

**Comment by tecnobrat** [comment]
@dharrisio the environment.

**Comment by stack72** [comment]
This was fixed in #7671
